### PR TITLE
feat: add javascript intellisense support in vscode

### PR DESCRIPTION
**Pull request by shiv19**
If this PR is merged,
when users import with paths starting from ~ they will get intellisense.

### CODE REVIEW

**Comment by shiv19** [comment]
Caveat,
node_modules intellisense isn't working after this. 
(I tried adding `"*": ["./node_modules/tns-core-modules/*", "./node_modules/nativescript-*"],` to paths map)

File: jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "compilerOptions": {
+    "target": "es2017",
```
Comment (manoldonev) [inline jsconfig.json]: Not sure about this value for the `target` property. Documentation says `target` "specifies which default library (lib.d.ts) to use" and by default (no webpack) we are currently targeting es5.

I would assume you are using a webpack-only development workflow (with es6 modules?) and that is why you specified higher target here?

File: jsconfig.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    "allowSyntheticDefaultImports": false,
+    "baseUrl": "./",
+    "paths": {
+      "~/*": ["app/*"]
+    }
+  },
+  "exclude": ["node_modules", "platforms", "hooks"]
```
Comment (manoldonev) [inline jsconfig.json]: [Docs](https://code.visualstudio.com/docs/languages/javascript#_javascript-projects-jsconfigjson) suggest as a best practice to use a single "include" instead:
``` js
    "include": ["app/**/*"]
```

**Comment by manoldonev** [comment]
> node_modules intellisense isn't working after this.

Could you elaborate about this caveat -- what is not working?

File: jsconfig.json (hunk 1 above)
Comment (manoldonev) [reply jsconfig.json]: @shiv19 auto-complete seems to have stopped working on my end without `target` attribute. Can you check? (make sure you restart vscode when testing with/without `target`).

**Comment by shiv19** [comment]
@manoldonev 
Tried restarting VS Code.
Without the target, it works fine for me.

I'm getting proper code completion for imports starting with ~/, ../ and from node_modules

**Comment by manoldonev** [comment]
@shiv19 indeed you are right, tested again and it autocompletes most of the time (I guess no problem with the config, probably vscode issue).

**Comment by shiv19** [comment]
Thank you :)
I've submitted same PR for other js templates as well :)

**Comment by manoldonev** [comment]
@shiv19 merged them and updated the rest of the js templates as well 👍

**Pull request merged**